[DEVX-265] : Improve README.md of Python SDK (#203)
Clarifai/clarifai-python: README.md

[Website](https://www.clarifai.com/) | [Demo](https://clarifai.com/demo) | [Signup for a Free Account](https://clarifai.com/signup) | [API Docs](https://docs.clarifai.com/) | [Clarifai Community](https://clarifai.com/explore) | [Python SDK Docs](https://docs.clarifai.com/python-sdk/api-reference) | [Examples](https://github.com/Clarifai/examples) | [Colab Notebooks](https://github.com/Clarifai/colab-notebooks)

---

## Table Of Contents

* **[Installation](#installation)**
* **[Getting Started](#getting-started)**
* **[Interacting with Datasets](#interacting-with-datasets)**
* **[Interacting with Inputs](#interacting-with-inputs)**
  * [Input Upload](#input-upload)
  * [Input Listing](#input-listing)

* **[Interacting with Models](#interacting-with-models)**
  * [Workflow Predict](#workflow-predict)
  * [Workflow Listing](#workflows-listing)
  * [Workflow Create](#workflow-create)
  * [Workflow Export](#workflow-export)

> * **[More Examples](#more-examples)**

Reviewer: I quite like the sub-list @sainivedh . Are you referring to those?

> ## Table Of Contents
> 
> * **[Installation](#installation)**
> * **[Getting Started](#getting-started)**
> * **[Interacting with Datasets](#interacting-with-datasets)**
> * **[Interacting with Inputs](#interacting-with-inputs)**
>   * [Input Upload](#input-upload)
>   * [Input Listing](#input-listing)
> 
> * **[Interacting with Models](#interacting-with-models)**
>   * [Workflow Predict](#workflow-predict)
>   * [Workflow Listing](#workflows-listing)
>   * [Workflow Create](#workflow-create)
>   * [Workflow Export](#workflow-export)
> 
> * **[More Examples](#more-examples)**

Reviewer 2: This regarding the spacing between main bullet points
```suggestion
## Table Of Contents
- **[Installation](#installation)**
- **[Getting Started](#getting-started)**
- **[Interacting with Datasets](#interacting-with-datasets)**
- **[Interacting with Inputs](#interacting-with-inputs)**
  * [Input Upload](#input-upload)
  * [Input Listing](#input-listing)
- **[Interacting with Models](#interacting-with-models)**
  * [Workflow Predict](#workflow-predict)
  * [Workflow Listing](#workflows-listing)
  * [Workflow Create](#workflow-create)
  * [Workflow Export](#workflow-export)
- **[More Examples](#more-examples)**
```
![image](https://github.com/Clarifai/clarifai-python/assets/39689508/02804dc7-1f6b-4bed-8160-c6564509414f)

This regarding the spacing This regarding the spacing

---

> ##Table Of Contents

Reviewer: ```suggestion
## Table Of Contents
```

> **[1. Installation](#installation)** \

Reviewer 2: Either of numbering/bullet point should be good. I'm not sure if we need both. Could we use just bullet points

Author: I have made the changes.Can you check.

**[2. Getting Started](#getting-started)**
**[3. Interacting with Datasets](#interacting-with-datasets)**

> **[4. Interacting with Inputs](#interacting-with-inputs)**

Reviewer: Need a new line in between. Right now all 4 are on the same line.

Author: ![image](https://github.com/Clarifai/clarifai-python/assets/147609006/ae14c6a3-f76a-4494-8734-5209f5a51728)

They are on separate lines when previewed.

Reviewer: That's odd. Both my preview in VS Code and Github shows that they are on the same line....
![Screenshot 2023-10-30 at 11 52 04 AM](https://github.com/Clarifai/clarifai-python/assets/48971969/8baa2ff4-c3af-4d7b-a8c2-e6012bfd2c11)
^ VS Code

![Screenshot 2023-10-30 at 11 52 54 AM](https://github.com/Clarifai/clarifai-python/assets/48971969/4c67e79b-2ac6-43e1-94f1-6c1a861275a7)

^ Github. [Try click this link to see the preview in Github](https://github.com/Clarifai/clarifai-python/blob/a06ca909a198b7adcbc1da5079e1c2c81680713f/README.md).

Author: OK, I will check that out.

Author: I have made the changes. Can you check?

> * [4.1 Input upload](#input-upload)

Reviewer 2: ```suggestion
* [4.1 Input Upload](#input-upload)
```

> * [4.2 Input listing](#input-listing)

Reviewer 2: ```suggestion
* [4.2 Inputs Listing](#input-listing)
```

## Getting started
Clarifai uses **Personal Access Tokens(PATs)** to validate requests. You can create and manage PATs under your Clarifai account security settings.

[Create PAT:](https://docs.clarifai.com/clarifai-basics/authentication/personal-access-tokens/)
* ***Log into Portal &rarr; Profile Icon &rarr; Security Settings &rarr; Create Personal Access Token &rarr; Set the scopes &rarr; Confirm***

Export your PAT as an environment variable. Then, import and initialize the API Client.

[Get User ID:](https://help.clarifai.com/hc/en-us/articles/4408131912727-How-do-I-find-my-user-id-app-id-and-PAT-)

* ***Log into Portal &rarr; Profile Icon &rarr; Account &rarr; Profile &rarr; User-ID***

> Set PAT as environment variable through terminal:

Reviewer: Maybe make "Create PAT" as a bullet point, and put the description on the same line? 
Same for User ID. 
Then, group both bullets together above the line "Export your PAT...".

Author: ![image](https://github.com/Clarifai/clarifai-python/assets/147609006/d3510181-b9f0-4e4d-9fe8-99343ef702fb)

Is this fine?

Reviewer: Yes, thanks!

## Getting started
Clarifai uses **Personal Access Tokens(PATs)** to validate requests. You can create and manage PATs under your Clarifai account security settings.

* [Create PAT:](https://docs.clarifai.com/clarifai-basics/authentication/personal-access-tokens/) ***Log into Portal &rarr; Profile Icon &rarr; Security Settings &rarr; Create Personal Access Token &rarr; Set the scopes &rarr; Confirm***

> <br />

Reviewer: ![Screenshot 2023-10-30 at 1 16 06 PM](https://github.com/Clarifai/clarifai-python/assets/48971969/5bb7a6b2-3135-4787-b86c-41b1b79a912c)
Maybe remove this <br>? Looks a bit odd to have that much space between 2 bullets.

Reviewer: Otherwise, LGTM

Author: ok, I will change it.

Reviewer: Thanks! Looks good, but it seems like some linting tests are failing. Did you set up your pre-commit hooks?

Author: ![image](https://github.com/Clarifai/clarifai-python/assets/147609006/5d6d8641-7dbf-4c1d-8bfe-daffb9cb842e)

yes.

Reviewer: This looks good to me, thanks for iterating. Just need to figure out why the tests are failing and address @sainivedh 's comments :)

apps = list(apps_generator)
```

> ### 💾 Interacting with Datasets

Reviewer 2: ```suggestion
## 💾 Interacting with Datasets
```

```

> ### ⌨️ Interacting with Inputs

Reviewer 2: ```suggestion
##  💾 Interacting with Inputs
```
Inputs is also a data, so having keyboard icon this seems misleading wr.t python `input()`

```

> ### 🥁 Interacting with Models

Reviewer 2: ```suggestion
## 🥁 Interacting with Models
```

> Using Clarifai models, you can choose any models  by provding model URL or ID to perform prediction and  list all available models. It is also possible to modify model prediction outcomes by adding threshold.

Reviewer 2: Descriptions could be made more verbose and detailed.
```suggestion
The **Model** Class allows you to perform predictions using Clarifai models. You can specify which model to use by providing the model URL or ID. This gives you flexibility in choosing models. The **App** Class also allows listing all available Clarifai models for discovery. 

For greater control over model predictions, you can pass in an `output_config` to modify the model output as demonstrated below. 
```

all_llm_community_models = list(all_llm_community_models)
```

> ### 🔥 Interacting with Workflows

Reviewer 2: ```suggestion
## 🔥 Interacting with Workflows
```

> Using ***Workflow()***, you can interact with Calrifai workflows in many ways. It allows prediction and modification of inference thresholds to get desirable results. Creating and exporting of workflows can be done using local ***yaml*** file.

Reviewer 2: ```suggestion
Workflows offer a versatile framework for constructing the inference pipeline, simplifying the integration of diverse models. You can use the **Workflow** class to create and manage workflows using **YAML** configuration.

For starting or making quick adjustments to existing Clarifai community workflows using an initial YAML configuration, the SDK provides an export feature.
```
